File: scripts/component_graph/server/util/env.py
#!/usr/bin/env python3
# Copyright 2019 The Fuchsia Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.
""" Provides OS environment utility functionality. """

import os


def get_fuchsia_root():
    """Returns the fuchsia the from the environment."""
    fuchsia_dir = os.environ.get("FUCHSIA_DIR")
    if not os.path.exists(fuchsia_dir):
        return None
    if not fuchsia_dir.endswith("/"):
        fuchsia_dir += "/"
    return fuchsia_dir

File: src/devices/driver_manager/env.cc
// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "env.h"

#include <stdlib.h>
#include <string.h>

bool getenv_bool(const char* key, bool default_value) {
  const char* value = getenv(key);
  if (value == nullptr) {
    return default_value;
  }
  if ((strcmp(value, "0") == 0) || (strcmp(value, "false") == 0) || (strcmp(value, "off") == 0)) {
    return false;
  }
  return true;
}
